src/main/java/com/fundynamic/d2tm/game/map/Cell.java

```diff
@@ -7,10 +7,12 @@
 public class Cell {
 
     private Terrain terrain;
+    private boolean shrouded;
 
     public Cell(Terrain terrain) {
         if (terrain == null) throw new IllegalArgumentException("Terrain argument may not be null");
         this.terrain = terrain;
+        this.shrouded = true;
```

Comment:
this won't work. Especially since 'shroud' is just a flag if it is visible. There will probably be a distinction between the actual terrain data and who can see it. Perhaps it will become `shrouded[PLAYER_IDS]` or perhaps `shrouded` will move elsewhere later? We could keep it here for now. But once we take care of AI or 2 player games, etc, then it will become troublesome :)